### PR TITLE
Add a branch checkout in the image building step

### PR DESCRIPTION
**Pull request by maelk**
To build the local images from a branch if needed when cloned

For example :
IRONIC_LOCAL_IMAGE=https://github.com/metal3-io/ironic-image.git
IRONIC_LOCAL_IMAGE_BRANCH=master
to build the image from the master branch of the ironic repo (defaults to master)

### CODE REVIEW

**Comment by maelk** [comment]
This PR does not consider the cases when the repo is cloned by is not on the correct branch to follow what was done before, if the folder exists, we directly use it.

File: 02_configure_host.sh [hunk 1]
```diff
@@ -148,8 +150,14 @@ for IMAGE_VAR in $(env | grep "_LOCAL_IMAGE=" | grep -o "^[^=]*") ; do
   if [[ "$IMAGE" =~ "://" ]] ; then
     REPOPATH=~/${IMAGE##*/}
     # Clone to ~ if not there already
-    [ -e "$REPOPATH" ] || git clone "$IMAGE" "$REPOPATH"
-    cd "$REPOPATH" || exit
+    if [ -e "$REPOPATH" ]; then
+       cd "$REPOPATH" || exit
```
Comment (kanwar-saad) [inline 02_configure_host.sh]: please one kind of variable name either "$VAR" or $VAR or ${VAR} or "${VAR}". It is different everywhere now. I prefer "${VAR}"

File: 02_configure_host.sh [hunk 2]
```diff
@@ -148,8 +150,14 @@ for IMAGE_VAR in $(env | grep "_LOCAL_IMAGE=" | grep -o "^[^=]*") ; do
   if [[ "$IMAGE" =~ "://" ]] ; then
     REPOPATH=~/${IMAGE##*/}
     # Clone to ~ if not there already
-    [ -e "$REPOPATH" ] || git clone "$IMAGE" "$REPOPATH"
-    cd "$REPOPATH" || exit
+    if [ -e "$REPOPATH" ]; then
```
Comment (kanwar-saad) [inline 02_configure_host.sh]: it can be like this

if dir not exist
  then clone
end if

cd dir
switch branch

**Pull request merged**